library/src/blas3/Tensile/gemm_templates.cpp

```diff
@@ -76,61 +77,70 @@ rocblas_status rocblas_internal_gemm(rocblas_handle    handle,
+                                      offset_a,
+                                      ldb,
+                                      stride_b,
+                                      offset_b,
+                                      m,
+                                      n,
+                                      k,
+                                      batch_count);
     }
     else
     {
-        return rocblas_call_tensile(handle,
-                                    alpha,
-                                    beta,
-                                    A + offset_a,
-                                    B + offset_b,
-                                    C + offset_c,
-                                    C + offset_c,
-                                    trans_a,
-                                    trans_b,
-                                    ldc,
-                                    stride_c,
-                                    0,
-                                    ldc,
-                                    stride_c,
-                                    0,
-                                    lda,
-                                    stride_a,
-                                    0,
-                                    ldb,
-                                    stride_b,
-                                    0,
-                                    m,
-                                    n,
-                                    k,
-                                    batch_count);
+        status = rocblas_call_tensile(handle,
+                                      alpha,
+                                      beta,
+                                      A + offset_a,
+                                      B + offset_b,
+                                      C + offset_c,
+                                      C + offset_c,
+                                      trans_a,
+                                      trans_b,
+                                      ldc,
+                                      stride_c,
+                                      0,
+                                      ldc,
+                                      stride_c,
+                                      0,
+                                      lda,
+                                      stride_a,
+                                      0,
+                                      ldb,
+                                      stride_b,
+                                      0,
+                                      m,
+                                      n,
+                                      k,
+                                      batch_count);
+    }
+
+    // Return the current status if a solution was not found
+    // by other libraries(hipBLASLt or Tensile) or if the data is not both complex.
+    // Otherwise, fall through to rocblas source GEMM.
+    if(status != rocblas_status_not_implemented || !rocblas_is_complex<TScal>)
+    {
+        return status;
```

Comment:
Is the second logic condition  `!rocblas_is_complex<TScal>)` required? What is `rocblas_status` return from `rocblas_call_tensile` if it is called with `rocblas_is_complex<TScal>`?

Reply:
I recall the comment, I see the fall-through is only for complex.